datasets/swda/swda.py

```diff
@@ -83,6 +82,305 @@
+    "qy^g^r",
+    "no^r",
+    "qh(^q)",
+    "sd;sv",
+    "bf(^q)",
+    "+",
+    "qy^2",
+    "qw^d",
+    "qy^g",
+    "qh^g",
+    "nn^t",
+    "ad^r",
+    "oo^t",
+    "co^c",
+    "ng",
+    "^q",
+    "qw^d^c",
+    "qrr^t",
+    "^h",
+    "aap^r",
+    "bc^r",
+    "sd^m",
+    "bk^r",
+    "qy^g^c",
+    "qr(^q)",
+    "ng^t",
+    "arp",
+    "h",
+    "bh",
+    "sd^c",
+    "^g",
+    "o^r",
+    "qy^c",
+    "sd^e",
+    "fw",
+    "ar^r",
+    "qy^m",
+    "bc",
+    "sv^t",
+    "aap^m",
+    "sd;no",
+    "ng^r",
+    "bf^g",
+    "sd^e^t",
+    "o^c",
+    "b^r",
+    "b^m^g",
+    "ba",
+    "t1",
+    "qy^d(^q)",
+    "nn^m",
+    "ny",
+    "ba,fe",
+    "aa^m",
+    "qh",
+    "na^m",
+    "oo(^q)",
+    "qw^t",
+    "na^t",
+    "qh^h",
+    "qy^d^m",
+    "ny^m",
+    "fa",
+    "qy^d",
+    "fc^t",
+    "sd(^q)",
+    "qy^d^r",
+    "bf^m",
+    "sd(^q)^t",
+    "ft^t",
+    "^q^r",
+    "sd^t",
+    "sd(^q)^r",
+    "ad^t",
+]
+
+# Damsl dialogue act tags version - short version 43 dialogue acts labels.
+damsl_act_tag = [
+    "ad",
+    "qo",
+    "qy",
+    "arp_nd",
+    "sd",
+    "h",
+    "bh",
+    "no",
+    "^2",
+    "^g",
+    "ar",
+    "aa",
+    "sv",
+    "bk",
+    "fp",
+    "qw",
+    "b",
+    "ba",
+    "t1",
+    "oo_co_cc",
+    "+",
+    "ny",
+    "qw^d",
+    "x",
+    "qh",
+    "fc",
+    'fo_o_fw_"_by_bc',
+    "aap_am",
+    "%",
+    "bf",
+    "t3",
+    "nn",
+    "bd",
+    "ng",
+    "^q",
+    "br",
+    "qy^d",
+    "fa",
+    "^h",
+    "b^m",
+    "ft",
+    "qrr",
+    "na",
+]
+
+# Data features types using PyArrow data types.
+data_features_types = {
+    "swda_filename": datasets.Value("string"),
+    "ptb_basename": datasets.Value("string"),
+    "conversation_no": datasets.Value("int64"),
+    "transcript_index": datasets.Value("int64"),
+    "act_tag": datasets.ClassLabel(num_classes=217, names=act_tags),
+    "damsl_act_tag": datasets.ClassLabel(num_classes=43, names=damsl_act_tag),
+    "caller": datasets.Value("string"),
+    "utterance_index": datasets.Value("int64"),
+    "subutterance_index": datasets.Value("int64"),
+    "text": datasets.Value("string"),
+    "pos": datasets.Value("string"),
+    "trees": datasets.Value("string"),
+    "ptb_treenumbers": datasets.Value("string"),
+    "talk_day": datasets.Value("string"),
+    "length": datasets.Value("int64"),
+    "topic_description": datasets.Value("string"),
+    "prompt": datasets.Value("string"),
+    "from_caller": datasets.Value("int64"),
+    "from_caller_sex": datasets.Value("string"),
+    "from_caller_education": datasets.Value("int64"),
+    "from_caller_birth_year": datasets.Value("int64"),
+    "from_caller_dialect_area": datasets.Value("string"),
+    "to_caller": datasets.Value("int64"),
+    "to_caller_sex": datasets.Value("string"),
+    "to_caller_education": datasets.Value("int64"),
+    "to_caller_birth_year": datasets.Value("int64"),
+    "to_caller_dialect_area": datasets.Value("string"),
+}
```

Comment:
I'd rather have the features defined directly in the `_info` function to be more consistent with other datasets, not sure whether there are any benefits to having them here.

Reply:
@yjernite  Thank you for all the comments and suggestions! 

You are correct, it isn't consistent at all with other datasets that have all this in `_info`, I hope you can help me out here 🙏🏼  : 
I desperately want to avoid having this features dictionary(str, datasets.Value()) data type written twice (once in `_info` and once in `_generate_examples` - where I only need use the keys)...

I want a way to access the feature names declared in `_info` into `_generate_examples` so I can nicely do `yield id_, {feature: utt[feature] for feature in data_features_types.keys()}`.

What do you think would be the most elegant way of doing this?

For the sake of simplicity I can have the feature names copied again in `_generate_examples ` but there's gotta be a better way 😥.

Sorry, the perfectionist in me is struggling 😓.

Reply:
You can access the features in _generate_examples via `self.info.features` :)

Reply:
Thanks @lhoestq! Sorry for the slow reply!

Reply:
Wow, it was right under my nose 😄 Thank you @lhoestq !